### PR TITLE
Add BYOL with syncbn and accumulated gradients

### PR DESCRIPTION
**Pull request by scnuhealthy**

### CODE REVIEW

File: openselfsup/models/byol.py [hunk 1]
```diff
@@ -70,11 +70,7 @@ def _momentum_update(self):
             param_tgt.data = param_tgt.data * self.momentum + \
                              param_ol.data * (1. - self.momentum)
 
-    @torch.no_grad()
-    def momentum_update(self):
-        self._momentum_update()
-
-    def forward_train(self, img, **kwargs):
+    def forward_train(self, img, weight=None, **kwargs):
```
Comment (XiaohangZhan) [inline openselfsup/models/byol.py]: What is the `weight`?

File: configs/selfsup/byol/r50_bs4096_ep300_syncbn_grad_accmulate.py [hunk 2]
```diff
@@ -79,16 +78,16 @@
         pipeline2=train_pipeline2))
 # additional hooks
 custom_hooks = [
-    dict(type='BYOLHook', end_momentum=1.)
+    dict(type='BYOLHook', end_momentum=1., update_interval=16)
```
Comment (XiaohangZhan) [inline configs/selfsup/byol/r50_bs4096_ep300_syncbn_grad_accmulate.py:81]: You'd better merge the two `update_interval` since they bind to each other. In this way, when some users want to change the update_interval, it will be changed both in two positions.
E.g.,
`update_interval=16 # graduate accumulation iterations for BYOL`
`custom_hooks = [
    dict(type='BYOLHook', end_momentum=1., update_interval=update_interval)]`
`optimizer_config = dict(update_interval=update_interval, use_fp16=use_fp16)`

File: openselfsup/hooks/byol_hook.py [hunk 3]
```diff
@@ -31,3 +33,10 @@ def before_train_iter(self, runner):
         m = self.end_momentum - (self.end_momentum - base_m) * (
             cos(pi * cur_iter / float(max_iter)) + 1) / 2
         runner.model.module.momentum = m
```
Comment (XiaohangZhan) [inline openselfsup/hooks/byol_hook.py:35]: Is update_interval applicable to the update of this momentum value?

File: docs/MODEL_ZOO.md [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
+<tr><td><a href="https://arxiv.org/abs/2006.07733" target="_blank" rel="noopener noreferrer">BYOL</a> (<a href="https://github.com/XiaohangZhan">@XiaohangZhan</a>)</td><td>selfsup/byol/r50_bs4096_ep200.py (<a href="https://github.com/xieenze">@xieenze</a>)</td><td>default</td><td><a href="https://drive.google.com/file/d/1Whj3j5E3ShQj_VufjrJSzWiq1xcZZCXN/view?usp=sharing" target="_blank" rel="noopener noreferrer">byol_r50-e3b0c442.pth</a></td><td>4096 (256 for speed test)</td><td>200</td><td>30min57s</td></tr>
+<tr><td><a href="https://arxiv.org/abs/2006.07733" target="_blank" rel="noopener noreferrer">BYOL</a>(<a href="https://github.com/XiaohangZhan">@scnuhealthy</a>)</td><td>selfsup/byol/r50_bs256_accumulate_ep300.py</td><td>default</td><td><a href="https://drive.google.com/file/d/12Zu9r3fE8qKF4OW6WQXa5Ec6VuA2m3j7/view?usp=sharing" target="_blank" rel="noopener noreferrer">byol_r50_bs256_accmulate16_ep300-5df46722.pth</a></td><td>256</td><td>300</td><td></td></tr>
```
Comment (XiaohangZhan) [inline docs/MODEL_ZOO.md:26]: Please also test the average epoch time. You may refer to your training log.

File: openselfsup/models/necks.py [hunk 5]
```diff
@@ -230,6 +230,89 @@ def forward(self, x):
         return [self.mlp(x.view(x.size(0), -1))]
 
 
+@NECKS.register_module
+class NonLinearNeckBYOL(nn.Module):
```
Comment (XiaohangZhan) [inline openselfsup/models/necks.py:234]: There is a lot of repeating codes between NonLinearNeckBYOL and NonLinearNeckSimCLR, please merge them into one class.

**Comment by scnuhealthy** [comment]
Ok, I will solve them and pull again
